File: app-java/src/main/java/net/mEmoZz/yts/java/ui/base/BasePresenter.java
package net.mEmoZz.yts.java.ui.base;

/**
 * Authored by Mohamed Fathy on 08 Mar, 2018.
 * Contact: muhamed.gendy@gmail.com
 */

public interface BasePresenter<V extends BaseView, I extends BaseInteractor> {

  void onAttach(V view, I interactor);

  void onDestroy();
}

File: app-kotlin/src/main/java/net/mEmoZz/yts/kotlin/ui/base/BasePresenter.kt
package net.mEmoZz.yts.kotlin.ui.base

/**
 * Authored by Mohamed Fathy on 08 Mar, 2018.
 * Contact: muhamed.gendy@gmail.com
 */

interface BasePresenter<in V, in I : BaseInteractor> {

  fun onAttach(view: V, interactor: I)

  fun onDestroy()
}